01_financial-well-being/income_metrics_all.qmd

```diff
@@ -0,0 +1,676 @@
+```
+
+Create a variable that assigns a weight to each place based on the total 2020 population. First, create a variable for the total population.
+
+```{r}
+puma_place_2021 <- puma_place_2021 %>%
+  mutate(totpop = sum(pop20))
+
+```
+
+Then, create a variable for the share of total population comprised by each place (unique state fip + county pairs)
+
+```{r}
+puma_place_2021 <- puma_place_2021 %>% 
+  group_by(statefip, place) %>% 
+  mutate(placepop = sum(pop20),
+                placewgt = placepop/totpop)
+
+```
+
+Look at distribution of place total population shares (unique). NYC, LA and Chicago all hold disproportionate shares of the population.
+
+```{r}
+
+puma_place_2021 %>%
+  select(statefip, place, placewgt) %>%
+  unique() %>%
+  ggplot(aes(x = placewgt)) +
+  geom_density(color = "blue", fill = alpha("blue", 0.3)) +
+  theme_minimal() +
+  ggtitle("Place pop weight") +
+  ylab("Density")
+
+
+```
+
+### Create quality variable
+
+Prepare for Data Quality flag (at end).
+
+Create flags in the PUMA-place crosswalk for high percentage of data from outside of place. Per Greg (for counties), 75% or more from the county - in this case, place - is good, below 35% is bad, in between is marginal. This is calculated by taking the product of percentage of PUMA in place and percentage of place in PUMA for each place-PUMA pairing, and summing across the place.
+
+Create new vars of interest.Products is the sum of the share of the place captured in the PUMA (afact) and the share of the PUMA captured in the place (afact2).
+
+```{r}
+
+puma_place <- puma_place_2021 %>%
+  mutate(products = afact*afact2)
+
+puma_place <- puma_place %>%
+  group_by(statefip, place) %>%
+  mutate(sum_products = sum(products),
+                place_pop = sum(pop20))
+
+
+```
+
+Sum_products mean is 0.76.
+Bottom percentile of population starts at 110629.
+
+```{r}
+summary(puma_place)
+```
+
+```{r}
+
+puma_place <- puma_place %>%
+  mutate(
+    puma_flag = 
+      case_when(
+        sum_products >= 0.75 ~ 1,
+        sum_products >= 0.35 ~ 2,
+        sum_products < 0.35 ~ 3
+      ),
+    small_place = 
+      case_when(
+        placepop >= 110629 ~ 0,
+        placepop < 110629 ~ 1
+      )
+  )
+
+```
+
+Save as "puma_place.csv" in the intermediate file.
+
+```{r}
+write_csv(puma_place, "data/intermediate/puma_place.csv")
+
+```
+
+Save a version with just the place-level values of data quality variables
+
+```{r}
+
+place_puma <- puma_place %>%
+  group_by(statefip, place) %>% 
+  summarise(puma_flag = mean(puma_flag), 
+            small_place = mean(small_place))
+
+```
+
+Save as "place_puma.csv" in the intermediate file.
+
+```{r}
+write_csv(place_puma, "data/intermediate/place_puma.csv")
+```
+
+### Prepare microdata
+
+First convert the IPUMS ACS missing values into NAs.
+
+Missing data is reported in a variety of ways - listed below:
+
+IPUMS says HHINCOME missing: 9999999 RENTGRS: No missing indicated OWNCOST: 99999 for not in universe
+
```

Comment:
Process question! Why are we working with rent and ownership costs in the income document? If we are going to load and process all of the ACS data, then maybe we need to put that in a standalone script?

Reply:
What about:

* `0` for `OWNERSHP`?
* `9999999` for `VALUEH`?
* `0` for `VACANCY`?
* `0` for `GRADEATT`?
* `0` for `EMPSTAT`? (For example, this question is only asked of people ages 16+).